File: hw/vizcaino/u3/hw26TemplateMethod/hw26TemplateMethodJavaScript/tea.js
const CaffeinBeverage = require('./caffeinBeverage');
const readline = require('readline');

class Tea extends CaffeinBeverage {
    brew() {
        console.log("Steep the tea");
    }

    addCondiments() {
        console.log("Adding lemon");
    }

    async wantsCondiments() {
        const answer = await this.getUserInput();
        return answer.toLowerCase().startsWith("y");
    }

    getUserInput() {
        return new Promise((resolve) => {
            const rl = readline.createInterface({
                input: process.stdin,
                output: process.stdout
            });
            rl.question("Would you like lemon with your tea (y/n)? ", (answer) => {
                rl.close();
                resolve(answer || "n");
            });
        });
    }
}

module.exports = Tea;

File: hw/vizcaino/u3/hw26TemplateMethod/hw26TemplateMethodPython/tea.py
from caffein_beverage import CaffeinBeverage

class Tea(CaffeinBeverage):
    def brew(self):
        print("Steep the tea")

    def add_condiments(self):
        print("Adding lemon")

    def wants_condiments(self):
        answer = self.get_user_input()
        return answer.lower().startswith("y")

    def get_user_input(self):
        try:
            answer = input("Would you like lemon with your tea (y/n)? ")
        except (IOError, KeyboardInterrupt):
            answer = "n"
        return answer if answer else "n"